docs/walk-through/dags.md

```diff
@@ -0,0 +1,234 @@
+# DAGs
+
+DAGs (Directed Acyclic Graphs) are similar to `Steps`, and the syntax to use DAGs is exactly the same. DAGs are formed
+of `Tasks`, and offer more flexibility in Workflow construction, with the key difference being that you can specify the
+dependencies of each `Task`, i.e. which other `Tasks` must run to completion before running this one, and Argo will
+construct the graph and run the Tasks in the desired order.
+
+## Specifying Dependencies
+
+The classic example for the `DAG` is the "diamond":
+
+
+=== "Hera"
+
+    ```py
+    from hera.workflows import DAG, Workflow, script
+
+
+    @script()
+    def echo(message):
+        print(message)
+
+
+    with Workflow(generate_name="dag-diamond-", entrypoint="diamond") as w:
+        with DAG(name="diamond"):
+            A = echo(name="A", arguments={"message": "A"})
+            B = echo(name="B", arguments={"message": "B"})
+            C = echo(name="C", arguments={"message": "C"})
+            D = echo(name="D", arguments={"message": "D"})
+
+            A >> [B, C] >> D
+    ```
+
+=== "YAML"
+
+    ```yaml
+    apiVersion: argoproj.io/v1alpha1
+    kind: Workflow
+    metadata:
+      generateName: dag-diamond-
+    spec:
+      entrypoint: diamond
+      templates:
+      - name: diamond
+        dag:
+          tasks:
+          - name: A
+            template: echo
+            arguments:
+              parameters:
+              - name: message
+                value: A
+          - name: B
+            depends: A
+            template: echo
+            arguments:
+              parameters:
+              - name: message
+                value: B
+          - name: C
+            depends: A
+            template: echo
+            arguments:
+              parameters:
+              - name: message
+                value: C
+          - name: D
+            depends: B && C
+            template: echo
+            arguments:
+              parameters:
+              - name: message
+                value: D
+      - name: echo
+        inputs:
+          parameters:
+          - name: message
+        script:
+          image: python:3.9
+          source: |-
+            import os
+            import sys
+            sys.path.append(os.getcwd())
+            import json
+            try: message = json.loads(r'''{{inputs.parameters.message}}''')
+            except: message = r'''{{inputs.parameters.message}}'''
+
+            print(message)
+          command:
+          - python
+
+    ```
+
+The rshift syntax (`>>`) is used with the returned objects from the `echo` calls; it specifies the left-hand-side of the
+operator is a dependency of the right-hand-side, i.e. `A >> B` means "B depends on A". This is syntactic sugar for
+`A.next(B)`, see the [Task dependencies explained](#task-dependencies-explained) section for more examples. A list acts
+as a boolean `and` of all the elements of the list, and it is important to note that a list cannot appear first in the
```

Comment:
maybe put in "(due to Python language limitations)" so people aren't left wondering why we don't support that?